### PR TITLE
explicit and implicit radiation fluxes labeled

### PR DESCRIPTION
**Pull request by kmdeck**
## Purpose 
In previous PRs, we made the following design choice:
"update cache" calls update aux for a model + update boundary fluxes for a model. It updates everything, regardless of if it is implicit or explicit.
"update implicit cache" updates implicit aux + implicit fluxes for a model ONLY.

Each step looks like this then:
- update entire cache, compute explicit tendency.
- each Newton iteration, update Y_imp + implicit cache, update implicit tendency, jacobian

The exception was with radiative fluxes. We only had a single function which computed SW and LW fluxes, but SW fluxes are treated explicitly, and only canopy LW is treated implicitly. 
For the canopy alone this was fine because canopy SW is independent of state. This is not the case in integrated models. For soil alone this was fine because everything was explicit, but this is not the case with integrated models with a canopy.

This led to an issue where we were overwriting the soil net radiation (for example) in the cache using canopy and soil temperatures at the next step, though the value at the previous step (explicit) was used to update the soil boundary conditions. This is not correct because then the value in the cache - which is used to set BC for the atmosphere - doesnt reflect what the soil actually saw.

We can correct this error by more clearly labeling LW as implicit and only updating it in the implicit boundary flux function. **This PR splits up SW and LW computations.**

It also moves the soil/snow integrated radiation computation outside of the component model's update boundary flux function. This makes it match the other integrated models (with a canopy), which require the "integrated model" to compute radiative transfer rather than computing each components net radiation individually.

## Content

----
- [ ] I have read and checked the items on the review checklist.

### CODE REVIEW

File: src/standalone/Vegetation/canopy_boundary_fluxes.jl [hunk 1]
```diff
@@ -139,7 +139,16 @@ function canopy_boundary_fluxes!(
     bc = canopy.boundary_conditions
 
     # Update the canopy radiation
-    canopy_radiant_energy_fluxes!(
+    canopy_shortwave_fluxes!(
```
[the remark this answers is not in the document]
Comment (kmdeck) [reply src/standalone/Vegetation/canopy_boundary_fluxes.jl]: Not all of these fluxes are needed in the explicit tendency, but we have made it so "update aux + update boundary fluxes" updates the whole cache at the beginning of each step (and is called 1x per explicit tendency computation).

Then there is a smaller subset of the cache which is the implicit cache and that is updated only via update_implicit* functions

File: src/integrated/soil_snow_model.jl [hunk 2]
```diff
@@ -478,7 +504,7 @@ in ϑ_l, ρe_int but explicitly in θ_i.
 """
 @kwdef struct SoilSublimationwithSnow{FT} <:
               ClimaLand.Soil.AbstractSoilSource{FT}
-    explicit::Bool = false
+    explicit::Bool = true
```
Comment (kmdeck) [inline src/integrated/soil_snow_model.jl]: All soil fluxes at the boundary are explicit - this was not updated in a previous PR which meant that it was included as part of the implicit tendency (not no contribution to Jacobian)

File: src/standalone/Snow/boundary_fluxes.jl [hunk 3]
```diff
@@ -103,21 +103,19 @@ function snow_boundary_fluxes!(
     t,
 ) where {FT}
     bc = model.boundary_conditions
-
-    SW_net = @. lazy((p.snow.α_snow - 1) * p.drivers.SW_d) #match sign convention in ./shared_utilities/drivers.jl
+    net_sw_radiation!(p.snow.SW_n, bc.radiation, model, Y, p, t)
```
Comment (juliasloan25) [inline src/standalone/Snow/boundary_fluxes.jl]: Are SW and LW both still computed in this function because everything is explicit for standalone snow?

**Review by juliasloan25 — COMMENTED** [review]
This looks good! One thing I find a little confusing is why `update_aux/update_boundary_fluxes` still update both the explicit and implicit variables. E.g. could we replace `update_radiation_fluxes` with `explicit_radiation_fluxes` to mirror `implicit_radiation_fluxes`?

File: src/standalone/Vegetation/canopy_boundary_fluxes.jl (hunk 1 above)
[the remark this answers is not in the document]
Comment (kmdeck) [reply src/standalone/Vegetation/canopy_boundary_fluxes.jl]: I think it is beneficial to have a function that updates everything. In ClimaAtmos, this function calls "update exp, then update imp". We could do that, but it is a separate change

File: src/standalone/Snow/boundary_fluxes.jl (hunk 3 above)
Comment (kmdeck) [reply src/standalone/Snow/boundary_fluxes.jl:106]: yes!

File: src/standalone/Vegetation/radiation.jl [hunk 4]
```diff
@@ -239,7 +238,38 @@ function canopy_radiant_energy_fluxes!(
     nir_d = p.canopy.radiative_transfer.nir_d
     f_abs_par = p.canopy.radiative_transfer.par.abs
     f_abs_nir = p.canopy.radiative_transfer.nir.abs
-    @. p.canopy.radiative_transfer.SW_n = f_abs_par * par_d + f_abs_nir * nir_d
+    @. p.canopy.radiative_transfer.SW_n =
```
[the remark this answers is not in the document]
Comment (kmdeck) [reply src/standalone/Vegetation/radiation.jl]: let me double check. we have a difference in sign convention for the canopy vs soil/snow for radiative fluxes. I think I was trying to unify them here.

File: test/integrated/soil_snow.jl [hunk 5]
```diff
@@ -78,7 +79,7 @@ set_initial_cache!(p, Y, t)
 @test all(parent(p.snow.total_energy_flux) .≈ 0)
 @test all(parent(p.snow.total_water_flux) .≈ 0)
 # Make sure the boundary conditions match bare soil result
-set_soil_initial_cache! = make_set_initial_cache(land_model.soil)
+set_soil_initial_cache! = make_set_initial_cache(soil_model)
```
[the remark this answers is not in the document]
Comment (kmdeck) [reply test/integrated/soil_snow.jl:82]: I think the problem was that the soil model in `land_model` has :snow in its `prognostic_land_component` tuple, while it does not in the soil model made in standalone. Rhis PR moves the radiative computation outside of the `soil_update_boundary_fluxes` function in the integrated model. So, the soil standalone initial cache update matches the cache update of the land model BUT would not match the cache update of land_model.soil.

**Comment by kmdeck** [comment]
This PR is stale, and I merged a change moving the implicit part of radiation to a separate function. Only that piece is updated in the implicit cache now, #1757 

We may still need to split up LW and SW for snow and soil. TBD. If I need to, Ill do it in a different PR
